spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2568,4 +2568,21 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("arrays_overlap") {
```

Comment:
Could you add tests where one or both input expressions evaluate to null on some rows? This can be achieved with a CASE expression as in some of the other array tests

Reply:
+1 for more coverage

Reply:
Case Expression based test case has been added.